src/execution/index/art/art.cpp

```diff
@@ -204,6 +204,148 @@ void ART::GenerateKeys(DataChunk &input, vector<unique_ptr<Key>> &keys) {
 	}
 }
 
+struct BuildARTOffsets {
+	BuildARTOffsets(idx_t start_p, idx_t end_p, data_t byte_p) : start(start_p), end(end_p), byte(byte_p) {};
+	idx_t start;
+	idx_t end;
+	data_t byte;
+};
+
+void ART::Build(vector<unique_ptr<Key>> &keys, row_t *row_ids, Node *&node, idx_t start_idx, idx_t end_idx,
+                idx_t depth) {
+
+	// test if we reached a leaf
+	auto prefix_start = depth;
+	while (keys[start_idx]->len != depth && keys[start_idx]->data[depth] == keys[end_idx]->data[depth]) {
+		depth++;
+	}
```

Comment:
This checks for prefix matching over the array of keys, but it's rather hard to read.
maybe have the while condition named?